crawler/file_processing.py

```diff
@@ -983,31 +976,28 @@ def parse_and_format_row(self, row, line_number, seen_rows) -> Any:
         modified_row[FIELD_SOURCE] = self.centre_config["name"]
 
         # extract the barcode and well coordinate
-        barcode_regex = self.centre_config["barcode_regex"]
         barcode_field = self.centre_config["barcode_field"]
 
         modified_row[FIELD_PLATE_BARCODE] = None  # type: ignore
-        if modified_row.get(barcode_field) and barcode_regex:
+        if modified_row.get(barcode_field) and (barcode_regex := self.centre_config["barcode_regex"]):
             (
                 modified_row[FIELD_PLATE_BARCODE],
                 modified_row[FIELD_COORDINATE],
             ) = self.extract_plate_barcode_and_coordinate(modified_row, line_number, barcode_field, barcode_regex)
+
         if not modified_row.get(FIELD_PLATE_BARCODE):
             return None
 
-        # add file details and import timestamp
-        import_timestamp = self.get_now_timestamp()
-
```

Comment:
Why remove this? It will add to the processing time to have to re-determine the current datetime for each of the three fields below. And there is no obvious reason we wouldn't expect these to be (very nearly) the same

Reply:
I'm thinking hard why I did... 🤔  I think, it seems like adding extra complexity to just wrap a one liner standard library call (`datetime.now()`) and they seem to be used in sequential lines so the time difference is negligible. In terms of overhead, I don't think 2 extra calls would would make a noticeable difference?

Reply:
I'm happy to keep the change now you've added it, especially with the freezegun plugin testing solution